ElectronicObserver/Window/Tools/SortieRecordViewer/Sortie/Battle/BattleFleets.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using ElectronicObserver.KancolleApi.Types.ApiGetMember.ShipDeck;
+using ElectronicObserver.KancolleApi.Types.Models;
+using ElectronicObserverTypes;
+using ElectronicObserverTypes.Mocks;
+
+namespace ElectronicObserver.Window.Tools.SortieRecordViewer.Sortie.Battle;
+
+public class BattleFleets
+{
+	public IFleetData Fleet { get; }
+	public IFleetData? EscortFleet { get; }
+	public List<IFleetData?>? Fleets { get; }
+	public List<IBaseAirCorpsData> AirBases { get; }
+	public IFleetData? FriendFleet { get; set; }
+	public IFleetData? EnemyFleet { get; set; }
+	public IFleetData? EnemyEscortFleet { get; set; }
+
+	private int CombinedFleetMainFleetShipCount => 6;
+
+	public BattleFleets(IFleetData fleet, IFleetData? escortFleet = null, List<IFleetData?>? fleets = null,
+		List<IBaseAirCorpsData>? airBases = null)
+	{
+		Fleet = fleet;
+		EscortFleet = escortFleet;
+		Fleets = fleets;
+		AirBases = airBases ?? new();
+	}
+
+	public BattleFleets Clone() => new(CloneFleet(Fleet), CloneFleet(EscortFleet), Fleets, AirBases.Select(CloneAirBase).ToList()!)
+	{
+		EnemyFleet = CloneFleet(EnemyFleet),
+		EnemyEscortFleet = CloneFleet(EnemyEscortFleet),
+	};
+
+	[return: NotNullIfNotNull(nameof(fleet))]
+	private static IFleetData? CloneFleet(IFleetData? fleet) => fleet switch
+	{
+		null => null,
+		_ => fleet.DeepClone(),
+	};
+
+	[return: NotNullIfNotNull(nameof(ab))]
+	private static IBaseAirCorpsData? CloneAirBase(IBaseAirCorpsData? ab) => ab switch
+	{
+		null => null,
+		_ => ab.DeepClone(),
+	};
+
+	public IShipData? GetShip(BattleIndex index) => index.FleetFlag switch
```

Comment:
This seems to not work with strike force

Reply:
fixed